### PR TITLE
Solved some Critical Issues in Formatting

### PR DESCRIPTION
**Pull request by AliMuhammadOfficial**

### CODE REVIEW

File: opengeode/Asn1scc.py [hunk 1]
```diff
@@ -81,11 +80,11 @@ def parse_asn1(*files, **options):
     project_cache = os.getenv ("PROJECT_CACHE")
     if project_cache is not None and not os.path.isdir(project_cache):
         try:
-            print("[INFO] Creating cache folder {}".format(project_cache))
+            print(f"[INFO] Creating cache folder {project_cache}")
             os.makedirs(project_cache)
         except OSError:
-            raise TypeError ("The configured cache folder \""
-            + project_cache + "\" is not there and could not be created\n")
+            raise TypeError (f'''The configured cache folder "{ project_cache} " \
```
Comment (maxime-esa) [inline opengeode/Asn1scc.py:86]: remove the space before the bracket {project_cache}

**Review by maxime-esa — CHANGES_REQUESTED** [review]

File: opengeode/AdaGenerator.py [hunk 2]
```diff
@@ -434,9 +434,7 @@ def rec_detect_choice(expr: ogAST.Expression) -> bool:
             continue
         context_elems.append(f'{var_name.lower()} {type_name(var_type, False)}')
 
-    asn1_context = (f'\n{process_asn1}-Context ::='' SEQUENCE {\n'
-            + ",\n   ".join(line.replace("_", "-").replace("'", '"') for line in context_elems)
-            + "\n}\n")
+    asn1_context = (f'''\n{process_asn1}-Context ::='' SEQUENCE {'\n,\n   '.join(line.replace("_", "-").replace("'", '"') for line in context_elems)}\n\n''')
```
Comment (maxime-esa) [inline opengeode/AdaGenerator.py:437]: I don't think this will work, did you test it ? If I recall correctly in f-strings you cannot do a '\n'.join(...)

**Review by maxime-esa — CHANGES_REQUESTED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Comment by AliMuhammadOfficial** [comment]
@maxime-esa should I create a pull request to implement unit testing for this repo?

File: opengeode/AdaGenerator.py [hunk 3]
```diff
@@ -434,9 +434,7 @@ def rec_detect_choice(expr: ogAST.Expression) -> bool:
             continue
         context_elems.append(f'{var_name.lower()} {type_name(var_type, False)}')
 
-    asn1_context = (f'\n{process_asn1}-Context ::='' SEQUENCE {\n'
-            + ",\n   ".join(line.replace("_", "-").replace("'", '"') for line in context_elems)
-            + "\n}\n")
+    asn1_context = ("\n{}-Context ::= SEQUENCE {}\n\n".format("hello", "\n,\n   ".join(line.replace("_", "-").replace("'", '"') for line in context_elems)))
```
Comment (maxime-esa) [inline opengeode/AdaGenerator.py:437]: Why is this still there? I thought you had reverted it?

File: opengeode/AdaGenerator.py [hunk 4]
```diff
@@ -469,8 +467,8 @@ def rec_detect_choice(expr: ogAST.Expression) -> bool:
             fromMod = f'{choiceTypeModule}.{ASN1SCC}{sortAda}'
             toMod = f'{process_name}_Datamodel.{ASN1SCC}{sortAda}'
             choice_selections.append(
-                    f'function To_{sortAda} (Src : {fromMod}) return {toMod} '
-                    f"is ({toMod}'Enum_Val (Src'Enum_Rep));")
+                    f"function To_{sortAda} (Src : {fromMod}) return {toMod} is \
+                        ({toMod}'Enum_Val (Src'Enum_Rep));")
```
Comment (maxime-esa) [inline opengeode/AdaGenerator.py:471]: Not a big fan of \ at the end of the lines, I prefer to keep it as it was

**Review by maxime-esa — CHANGES_REQUESTED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

File: opengeode/AdaGenerator.py (hunk 3 above)
Comment (AliMuhammadOfficial) [reply opengeode/AdaGenerator.py]: I really apologize for this. I changed butt not added in this commit I added code of my test version!

**Comment by maxime-esa** [comment]
Thanks! About the unit tests, what do you have in mind? If the plan is to add some py.tests (in tests/pytests folder) then I fully approve

**Comment by AliMuhammadOfficial** [comment]
Yes I will implement tests with pytest if you can create a task list and
set priorities it will help a lot!

**Pull request merged**